pandas/tests/io/test_orc.py

```diff
@@ -224,3 +224,24 @@ def test_orc_reader_snappy_compressed(dirpath):
     got = read_orc(inputfile).iloc[:10]
 
     tm.assert_equal(expected, got)
+
+
+def test_orc_roundtrip(dirpath):
```

Comment:
Will need a skip/xfail for older pyarrow versions, see the failing macos test which seems to use an older version of pyarrow.

Reply:
Sure. At the same time maybe we can update the pyarrow version in tests?

Reply:
If an older pyarrow version is useful for a different part of pandas that uses pyarrow (parquet maybe?) then it would be good not to bump the minimum version but to skip the test instead.

Reply:
Fixed.